nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/AnnotationService.java

```diff
@@ -44,37 +46,40 @@
      *
      * @param session the core session
      * @param annotationId the annotation document model id
+     * @throws NuxeoException if no annotation was found with the given id
      * @return the annotation, or {@code null} if the annotation with the given id was not found
      */
-    Annotation getAnnotation(CoreSession session, String annotationId);
+    Annotation getAnnotation(CoreSession session, String annotationId) throws NuxeoException;
```

Comment:
`NuxeoException` is a runtime and generic exception, there's no point in mentioning it as thrown. I don't like that the contract of the method rely on a generic exception. If you really want to specify something thrown, with application-specific meaning, then it should be a dedicated subclass of `NuxeoException`, like a new `AnnotationNotFoundException`.